### ISSUE TITLE
`removeError` always remove `filled` class

### ISSUE THREAD

**Issue by dcarral** [guillaumepotier/Parsley.js#1085]
`ParsleyUI.Field.removeError` method **always** removes the `filled` class from the `$errorsWrapper` element (see [source code](https://github.com/guillaumepotier/Parsley.js/blob/41362059bdb950a81dfb4536d33e7bd231f69985/src/parsley/ui.js#L215)).

Shouldn't it remove it **only if** the `Field` doesn't contain any more errors?

**Comment by marcandre** [comment]
You're right, it should. There's the fact that the `parsley-error` class on the input depends on `validationResult`, so you could `removeError`, still have another error (e.g. added manually, but that didn't change `validationResult`) and the `parsley-error` class would disappear.

The thing is: I don't quite see a way to have an API that would be consistent to manually add/remove/update errors.

This makes me think even more that `addError` and al. should be deprecated and not used.
